inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+        else                                                                                                                                                                                 \
+        {                                                                                                                                                                                    \
+            if (two_d_array->row_arrays[row_index] == NULL)                                                                                                                                  \
+            {                                                                                                                                                                                \
+                /* Codes_SRS_TWO_D_ARRAY_07_017: [ Otherwise, TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall allocate memory for the new row and return zero on success. ]*/                          \
+                unsigned char* new_row = malloc_2(two_d_array->cols, sizeof(T));                                                                                                             \
+                if (new_row == NULL)                                                                                                                                                         \
+                {                                                                                                                                                                            \
+                    /* Codes_SRS_TWO_D_ARRAY_07_018: [ If there are any errors then TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/                              \
+                    LogError("malloc(sizeof(col)=%zu) failed", two_d_array->cols * sizeof(T));                                                                                               \
+                    result = MU_FAILURE;                                                                                                                                                     \
+                }                                                                                                                                                                            \
+                else                                                                                                                                                                         \
+                {                                                                                                                                                                            \
+                    (void)memcpy((void*)&two_d_array->row_arrays[row_index], &new_row, sizeof(T*));                                                                                          \
```

Comment:
>(void)memcpy((void*)&two_d_array->row_arrays[row_index], &new_row, sizeof(T*));  [](http://example.com/codeflow?start=20&length=80)

I don't think we need to call memcpy to set the row, do we?

Reply:
removed memcpy